packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx

```diff
@@ -130,7 +131,7 @@ function Cell(props) {
           event.stopPropagation();
           event.preventDefault();
         } else if (onCellSelect) {
-          onCellSelect({ rowId, columnId }, { row: rowIndex, col: columnIndex });
+          onCellSelect({ rowId, columnId }, { row: rowIndex, col: columnIndex }, !!event.shiftKey, true);
```

Comment:
It seems like you have too many unnamed objects here.  With having four arguments of multiple objects, it might be easier to create an object with named properties and have the objects as values.  That would probably be easier to maintain and follow.  Instead of returning whether the shift key was pressed, I would return the actual event as a separate argument.  So something like onCellSelect(event, object)

Reply:
Created an object with all the arguments. I refrained from passing event from Cell to Row as Cell itself is handling it's needs.